app/scripts/controllers/main.js

```diff
@@ -186,6 +190,42 @@ angular.module('moodCatApp')
           list.scrollTop = list.scrollHeight;
         })
       }
+
+      $scope.setLatestMessage = function(gottenmessages) {
+        var biggest = -1;
+        for (var i = 0; i < gottenmessages.length; i++) {
+            if (gottenmessages[i].id > biggest) {
+             biggest = gottenmessages[i].id;
+            }
+        }
+        $scope.room.lastMessageId = biggest;
+      }
+
+      $scope.firstLatestMessage = function(){
+          $scope.setLatestMessage($scope.messages);
+      }
+
+      $scope.fetchMessages = function(){
```

Comment:
If possible, please bind this to `this` (the current controller) rather than `$scope`.